cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -388,17 +388,18 @@ func (c *client) GetMachinesForCluster(cluster *clusterv1.Cluster) ([]*clusterv1
 		return nil, errors.Wrapf(err, "error listing Machines for Cluster %s/%s", cluster.Namespace, cluster.Name)
 	}
 	var machines []*clusterv1.Machine
-	for _, m := range machineslist.Items {
+
+	for i := 0; i < len(machineslist.Items); i++ {
```

Comment:
Is there any reason you chose to write the `for` loop with index instead of using `range`?

Reply:
+1 on using range, but lets handle as a followup.

Reply:
Main reason was to be consistent with the other loops. I also remember that `range` reuses the same variable for each iteration, which might cause issues when passing pointers around.